File: backend/Domain/Entities/HRMS/LeaveBalance.cs
namespace Domain.Entities.HRMS;

using Domain.Common;

public class LeaveBalance : BaseEntity
{
    public int EmployeeId { get; set; }
    public Employee? Employee { get; set; }

    public int LeaveTypeId { get; set; }
    public LeaveType? LeaveType { get; set; }

    public decimal AllocatedHours { get; set; }
    public decimal UsedHours { get; set; }

    // Computed — not stored in DB
    public decimal RemainingHours => AllocatedHours - UsedHours;
}

File: dynamic-datatable-board/src/dto/hrms/leaveBalance.ts
// src/dto/hrms/leaveBalance.ts

export interface LeaveBalanceDto {
  id: number;
  employeeId: number;
  employeeName: string;
  leaveTypeId: number;
  leaveTypeName: string;
  allocatedHours: number;
  usedHours: number;
  remainingHours: number;
}
